### PR TITLE
[SYCL][libdevice] Add __imf_llmax/min, __imf_ullmax/min, __imf_umax/min

### PR DESCRIPTION
**Pull request by jinge90**
Signed-off-by: jinge90 <ge.jin@intel.com>

### CODE REVIEW

File: libdevice/device_imf.hpp [hunk 1]
```diff
@@ -95,13 +95,13 @@ static inline TyFP __integral2FP_host(TyINT x, int rdMode) {
 template <typename Ty> static inline Ty __imax(Ty x, Ty y) {
   static_assert(std::is_integral<Ty>::value,
                 "__imin only accepts integral type.");
-  return (x < y) ? x : y;
+  return (x > y) ? x : y;
 }
 
 template <typename Ty> static inline Ty __imin(Ty x, Ty y) {
   static_assert(std::is_integral<Ty>::value,
                 "__imax only accepts integral type.");
```
Comment (v-klochkov) [inline libdevice/device_imf.hpp:103]:
```suggestion
                "__imin only accepts integral type.");
```

File: libdevice/device_imf.hpp [hunk 2]
```diff
@@ -95,13 +95,13 @@ static inline TyFP __integral2FP_host(TyINT x, int rdMode) {
 template <typename Ty> static inline Ty __imax(Ty x, Ty y) {
   static_assert(std::is_integral<Ty>::value,
                 "__imin only accepts integral type.");
```
Comment (v-klochkov) [inline libdevice/device_imf.hpp:97]:
```suggestion
                "__imax only accepts integral type.");
```

File: libdevice/device_imf.hpp (hunk 1 above)
Comment (jinge90) [reply libdevice/device_imf.hpp]: Done.

**Comment by jinge90** [comment]
Hi, @intel/dpcpp-tools-reviewers 
Could you help review this PR?
Thanks very much.

**Comment by jinge90** [comment]
Hi, @pvchupin 
Could you take a look at this PR?
Thanks very much.

**Comment by asudarsa** [comment]
Hi @jinge90 

Can you please provide a 1-2 line description for this change?

Thanks

**Comment by xtian-github** [comment]
> Hi @jinge90
> 
> Can you please provide a 1-2 line description for this change?
> 
> Thanks

These math functions/operations are implemented to match CUDA math functions/operations for SYCL and OpenMP offloading on GPU.  Which is part of DSE OKR.

**Review by asudarsa — APPROVED** [review]
LGTM. Please add tests if possible. Thanks

**Comment by jinge90** [comment]
> LGTM. Please add tests if possible. Thanks

Hi, @asudarsa 
The test is here in review: https://github.com/intel/llvm-test-suite/pull/1275

Thanks very much.

**Review by xtian-github — APPROVED** [review]
LGTM

**Pull request merged**